### ISSUE TITLE
Support for 4.03's inline records for constructor arguments

### ISSUE THREAD

**Issue by hcarty** [whitequark/ppx_deriving_yojson#44]
``` ocaml
type t =
  | A of { x : int; y : int }
  | B of { c : string; d : float }
  [@@deriving yojson]
```

as referenced from #43

**Comment by whitequark** [comment]
It's unclear what sort of JSON should this correspond to. I think `{"type": "A", "x": 1, "y": 2}` etc is the most straightforward option.

In any case I don't have time to implement that.

**Comment by hcarty** [comment]
What do you think of `"constructor"` or `"label"` instead of `"type"`?

**Comment by whitequark** [comment]
I don't have an opinion one way or another. One name makes sense for OCaml, another for how JSON is usually expected to be shaped.

**Comment by whitequark** [comment]
Fixed in #50.
